core/src/smartcontracts/isi/query.rs

```diff
@@ -56,9 +56,9 @@ impl_lazy! {
 }
 
 /// Query Request statefully validated on the Iroha node side.
-#[derive(Debug, Decode, Encode)]
+#[derive(Debug, Clone, Decode, Encode)]
 #[repr(transparent)]
-pub struct ValidQueryRequest(SignedQuery);
+pub struct ValidQueryRequest(pub SignedQuery);
```

Comment:
Indeed, I just needed a getter, but created a way to break our state machine

Reply:
try to avoid a getter, that is precisely what would break state machine. Maybe you can implement `From<SignedQuery> for ValidQueryRequest` or have a method on `ValidQueryRequest` that does the job internally?

Reply:
This change is no logner needed, so I removed it